components/hab/src/cli.rs

```diff
@@ -1755,6 +1762,46 @@ mod tests {
             assert_eq!(error.kind, clap::ErrorKind::ValueValidation);
         }
 
+        #[test]
+        fn site_option_must_take_a_value() {
+            let matches = sub_sup_run(event_stream_enabled()).get_matches_from_safe(vec![
+                "run",
+                "--event-stream-application",
+                "MY_APP",
+                "--event-stream-environment",
+                "MY_ENV",
+                "--event-stream-token",
+                "MY_TOKEN",
+                "--event-stream-url",
+                "127.0.0.1:4222",
+                "--event-stream-site",
+            ]);
+            assert!(matches.is_err());
+            let error = matches.unwrap_err();
+            assert_eq!(error.kind, clap::ErrorKind::EmptyValue);
+            assert_eq!(error.info, Some(vec!["EVENT_STREAM_SITE".to_string()]));
+        }
+
+        #[test]
+        fn site_option_cannot_be_empty() {
+            let matches = sub_sup_run(event_stream_enabled()).get_matches_from_safe(vec![
+                "run",
+                "--event-stream-application",
+                "MY_APP",
+                "--event-stream-environment",
+                "MY_ENV",
+                "--event-stream-token",
+                "MY_TOKEN",
+                "--event-stream-url",
+                "127.0.0.1:4222",
+                "--event-stream-site",
+                "",
+            ]);
+            assert!(matches.is_err());
+            let error = matches.unwrap_err();
+            assert_eq!(error.kind, clap::ErrorKind::ValueValidation);
+        }
```

Comment:
It's awesome that there are tests!

A couple of suggestions for making them more focused and robust:
1. Since you're only testing one arg, extract the definition of `Arg::with_name("EVENT_STREAM_SITE")` to it's own function so you can focus on what you're testing and not all the setup required. There are 17 lines in each test and 10 of them aren't really relevant to what's being tested.
1. Try to avoid over-constraining your tests to details of clap that aren't relevant. Should that library change the name of structure of its errors, this test will break, but all we really care about is that `--event-stream-site` requires a non-empty value, so asserting that getting matches results in an error should be sufficient.
1. If we're testing that no value or an empty value is an error, we should also be testing that a non-empty value and omitting the option entirely are non-errors.

By the way, I realize you're following the template of the other tests that are already here, but I haven't reviewed those previous or I missed them.